bftengine/src/bftengine/RequestHandler.cpp

```diff
@@ -66,6 +67,25 @@ void RequestHandler::execute(IRequestsHandler::ExecutionRequestsQueue& requests,
       // Backward compatible with read only flag prior BC-5126
       req.flags = READ_ONLY_FLAG;
     }
+    // Replicas can publish an object e.g public_keys, configuration file, etc
+    // this object pass consensus, and replicas can perform action against is as:
+    // - validated that is equal to the object that is stored in memory of the replica.
+    // - save it to reserved pages.
+    // - proxy it to the application command handler.
+    if (req.flags & bftEngine::MsgFlag::CLIENTS_PUB_KEYS_FLAG) {
+      std::string recieved_keys(req.request, req.requestSize);
+      std::optional<std::string> bootstrap_keys;
+      if (req.flags & bftEngine::MsgFlag::PUBLISH_ON_CHAIN_OBJECT_FLAG) {
```

Comment:
is PUBLISH_ON_CHAIN_OBJECT_FLAG for initial exchange only?

Reply:
for publish keys, yes